controllers/machineset_controller.go

```diff
@@ -226,6 +227,28 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 		filteredMachines = append(filteredMachines, machine)
 	}
 
+	var errs []error
+	for _, machine := range filteredMachines {
+		if conditions.IsFalse(machine, clusterv1.MachineOwnerRemediatedCondition) {
+			logger.Info("Deleting unhealthy machine", "machine", machine.GetName())
+			patch := client.MergeFrom(machine.DeepCopy())
+			if err := r.Client.Delete(ctx, machine); err != nil {
+				errs = append(errs, errors.Wrap(err, "failed to delete"))
+				continue
+			}
+			conditions.MarkTrue(machine, clusterv1.MachineOwnerRemediatedCondition)
+			if err := r.Client.Status().Patch(ctx, machine, patch); err != nil {
+				errs = append(errs, errors.Wrap(err, "failed to update status"))
+			}
```

Comment:
This can potentially error in case the Machine deleted quickly and a NotFound error is returned, we should make sure to not log/return any errors if that happens.